docs/modules/xp.md

```diff
@@ -16,451 +16,1155 @@ You can find below all the function available in this module.
+    function #bs.xp:set_points
+
+    # See the result
+    # look at your XP bar in survival mode
+    ```
+:::
+:::{tab-item} Score
+
+**`#bs.xp:set_points`**
+
+Set the player points.
+
+Inputs
+
+:   (execution) `as <players>`
+    : The players to set points from.
+
+    (score) `$xp.set_points.points bs.in`: int
+    : The amount of points to set.
+
+Outputs
+
+:   (state) players xp
+    : The players XP is updated.
+
+Example
+
+:   Set your points to 1 using a score.
+
+    ```mcfunction
+    # Once (execute on you)
+    scoreboard players set $xp.set_points.points bs.in 1
+    function #bs.xp:set_points
+
+    # See the result
+    # look at your XP bar in survival mode
+    ```
+:::
+::::
+:::::
+:::::{tab-item} Total points
+::::{tab-set}
+:::{tab-item} Macro variable
+
+**`#bs.xp:set_total_points`**
+
+Set the total player points.
+
+Inputs
+
+:   (execution) `as <players>`
+    : The players you want to set the total XP.
+
+    (macro variable) `points`: int
+    : The amount of XP to set.
+
+Outputs
+
+:   (state) players xp
+    : The players XP is updated.
+
+Example
+
+:   Set your total XP amount to 42.
+
+    ```mcfunction
+    # Once (execute on you)
+    function #bs.xp:set_total_points {points:42}
+
+    # See the result
+    # look at your XP bar in survival mode
+    ```
+:::
+:::{tab-item} Storage
+
+**`#bs.xp:set_total_points`**
+
+Set the total player points.
+
+Inputs
+
+:   (execution) `as <players>`
+    : The players you want to set the total XP.
+
+    (storage) `bs:in xp.set_total_points.points`: int
+    : The amount of XP to set.
+
+Outputs
+
+:   (state) players xp
+    : The players XP is updated.
+
+Example
+
+:   Set your total XP amount to 61.
+
+    ```mcfunction
+    # Once (execute on you)
+    data modify storage bs:in xp.set_total_points.points set value 61
+    function #bs.xp:set_total_points
+
+    # See the result
+    # look at your XP bar in survival mode
+    ```
+:::
+:::{tab-item} Score
+
+**`#bs.xp:set_total_points`**
+
+Set the total player points.
+
+Inputs
+
+:   (execution) `as <players>`
+    : The players you want to set the total XP.
+
+    (score) `$xp.set_total_points.points bs.in`: int
+    : The amount of XP to set.
+
+Outputs
+
+:   (state) players xp
+    : The players XP is updated.
+
+Example
+
+:   Set your total XP amount to 1.
+
+    ```mcfunction
+    # Once (execute on you)
+    scoreboard players set $xp.set_total_points.points bs.in 1
+    function #bs.xp:set_total_points
+
+    # See the result
+    # look at your XP bar in survival mode
+    ```
+:::
+::::
+:::::
+:::::{tab-item} Progress bar
 ::::{tab-set}
-:::{tab-item} Points
+:::{tab-item} Macro variable
 
-**`bs.xp:add_points`**
+**`#bs.xp:set_progress`**
 
-Add XP from a score
+Fill partially the XP bar.
 
 Inputs
 
 :   (execution) `as <players>`
-    : The players to add XP to
+    : The players to fill the XP bar.
 
-    (score) `@s bs.in.0`
-    : The amount of XP to add
+    (macro variable) `points`: int
```

Comment:
"per_cent_mille" or "pcm" would be more consistent than "points" in the context of this function